From d181817db18a2172b616050e9d8d674d0d1ca003 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 10 Jan 2015 03:34:51 -0800
Subject: [PATCH 2005/2007] Update CheckArches to fallback on Intel/ppc if ld
 -v doesn't report supported architectures

Older versions of ld64 (eg: Xcode 3.x) do not report this.

http://www.llvm.org/bugs/show_bug.cgi?id=22182

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit b16ca706de5eeb244b1fe889760ae4339b23a170)
---
 make/platform/clang_darwin.mk | 2 ++
 1 file changed, 2 insertions(+)

diff --git llvm_release_35/projects/compiler-rt/make/platform/clang_darwin.mk macports_release_35/projects/compiler-rt/make/platform/clang_darwin.mk
index 6d3a0f8..2917c1e 100644
--- llvm_release_35/projects/compiler-rt/make/platform/clang_darwin.mk
+++ macports_release_35/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -25,6 +25,8 @@ CheckArches = \
           if $(LD) -v 2>&1 | grep "configured to support" \
              | tr ' ' '\n' | grep "^$$arch$$" >/dev/null 2>/dev/null; then \
             result="$$result$$arch "; \
+          elif ! ( $(LD) -v 2>&1 | grep -q "configured to support" ) && test "$$arch" = "i386" -o "$$arch" = "x86_64" -o "$$arch" = "ppc" -o "$$arch" = "ppc64" ; then \
+            result="$$result$$arch "; \
           else \
             printf 1>&2 \
             "warning: clang_darwin.mk: dropping arch '$$arch' from lib '$(2)'";\
-- 
2.2.2

